### PR TITLE
Battery icon

### PR DESCRIPTION
**Pull request by Topchris18**
I added an Batteryicon which is using the battery usermod of WLED to display the current battery status.
Screenshot_20241109_143216_WLED Native

This Icon only appears if the API is available otherwise it will be hidden.

### CODE REVIEW

File: gradle.properties [hunk 1]
```diff
@@ -9,4 +9,6 @@ android.useAndroidX=true
 android.nonTransitiveRClass=true
 
 #MPP
-kotlin.mpp.enableCInteropCommonization=true
\ No newline at end of file
+kotlin.mpp.enableCInteropCommonization=true
+
+ksp.incremental=false
```
Comment (Moustachauve) [inline gradle.properties]: Was this change intended?

File: app/src/main/res/drawable/baseline_battery_0_bar_24.xml [hunk 2]
```diff
@@ -0,0 +1,5 @@
+<vector xmlns:android="http://schemas.android.com/apk/res/android" android:height="24dp" android:tint="#000000" android:viewportHeight="24" android:viewportWidth="24" android:width="24dp">
```
Comment (Moustachauve) [inline app/src/main/res/drawable/baseline_battery_0_bar_24.xml:1]: I actually don't know if it makes a difference anymore, but I usually set the `android:tint` to `?attr/colorControlNormal`.

(Same in all the drawable)

File: app/src/main/java/ca/cgagnier/wlednativeandroid/ui/homeScreen/list/DeviceListItem.kt [hunk 3]
```diff
@@ -272,6 +272,18 @@ fun DeviceInfoTwoRows(
                         .height(20.dp)
                 )
             }
+
+            if(device.hasBattery){
```
Comment (Moustachauve) [inline app/src/main/java/ca/cgagnier/wlednativeandroid/ui/homeScreen/list/DeviceListItem.kt]: Nit: You're missing a space between the if and the condition. I recommend running the default code reformat on the code you changed to try to have a consistant style.

File: app/src/main/java/ca/cgagnier/wlednativeandroid/model/wledapi/Usermods.kt [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+import com.squareup.moshi.FromJson
+import com.squareup.moshi.Json
+import com.squareup.moshi.JsonClass
+import com.squareup.moshi.JsonQualifier
+
+
+@JsonClass(generateAdapter = true)
+data class Usermods (
+
+	@Json(name = "Battery level") val batLevel : List<Any>? = null,
```
Comment (Moustachauve) [inline app/src/main/java/ca/cgagnier/wlednativeandroid/model/wledapi/Usermods.kt]: nit: Don't abbreviate words in variable name. It should be `batteryLevel`.

Comment (Moustachauve) [inline app/src/main/java/ca/cgagnier/wlednativeandroid/model/wledapi/Usermods.kt]: I've not tried the battery usermod, I find it strange that it would have a list of battery level. Could you share with me an example of the JSON in the "u" section for the battery please?

I think that since usermods can be very varied, maybe there should be a package `package ca.cgagnier.wlednativeandroid.model.wledapi.usermods` that contains a class for each supported usermods. Then this usermod class would do this instead:

```
// In ca.cgagnier.wlednativeandroid.model.wledapi, UserMods.kt
@Json(name = "Battery level") val batteryLevel : BatteryLevel? = null,
```
```
// in ca.cgagnier.wlednativeandroid.model.wledapi.usermods, BatteryLevel.kt
@JsonClass(generateAdapter = true)
data class BatteryLevel (
[...]
```

File: app/src/main/java/ca/cgagnier/wlednativeandroid/model/wledapi/Info.kt [hunk 5]
```diff
@@ -47,4 +47,5 @@ data class Info (
     @Json(name = "product") val product : String? = null,
     @Json(name = "mac") val macAddress : String? = null,
     @Json(name = "ip") val ipAddress : String? = null,
+    @Json(name = "u") val usermods : Usermods? = null
```
Comment (Moustachauve) [inline app/src/main/java/ca/cgagnier/wlednativeandroid/model/wledapi/Info.kt]: Nit: should be `userMods` and `UserMods`

**Comment by Moustachauve** [comment]
Overall looks pretty good, thank you for the contribution! I left some comments, nothing that should be too major :)

File: app/src/main/java/ca/cgagnier/wlednativeandroid/model/wledapi/Usermods.kt (hunk 4 above)
Comment (Topchris18) [reply app/src/main/java/ca/cgagnier/wlednativeandroid/model/wledapi/Usermods.kt]:
``` json
"u": {
      "Battery level": [7, " %"],
      "Battery voltage": [4.27, " V"],
      "Next update": [0, " sec"],
      "AudioReactive": [
        "<button class=\"btn btn-xs\" onclick=\"requestJson({AudioReactive:{enabled:false}});\"><i class=\"icons on\">&#xe08f;</i></button>"
      ],
      "Audio Input Level": [
        "<div class=\"slider\"><div class=\"sliderwrap il\"><input class=\"noslide\" onchange=\"requestJson({AudioReactive:{inputLevel:parseInt(this.value)}});\" oninput=\"updateTrail(this);\" max=255 min=0 type=\"range\" value=128 /><div class=\"sliderdisplay\"></div></div></div>"
      ],
      "Audio Source": ["I2S digital", " - quiet"],
      "Sound Processing": ["running"],
      "Manual Gain": [1.56, "x"],
      "UDP Sound Sync": ["send mode", " v2"]
    },
```

This is the JSON of my usermods. As you can see there are no separated keys for each usermod.
BatteryLevel and voltage are an array because they include the unit of the number for whatever reason

File: gradle.properties (hunk 1 above)
Comment (Topchris18) [reply gradle.properties]: I removed it. I had problems with the database version and this was the first solution on the internet. Everything is working fine without it

File: app/src/main/res/drawable/baseline_battery_0_bar_24.xml (hunk 2 above)
Comment (Topchris18) [reply app/src/main/res/drawable/baseline_battery_0_bar_24.xml:1]: I just took those icons from MDI inside Android studio I surely can change them but this is how they were automatically imported
